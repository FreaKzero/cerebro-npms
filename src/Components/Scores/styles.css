.scores {
  display: flex;
  align-items: center;
  justify-content: center;
}

.scores > div {
  flex: 1;
  margin-right: 10px;
  padding: 5px;
  position: relative;
  text-align: center;
}

.scores > div > h4 {
  text-align: center;
  position: absolute;
  display: block;
  width: 100%;
  top: 36px;
  font-size: 16px;
  margin-left: -3px;
}

.scores > div > p {
  margin: 0;
  font-size: 14px;
}

.scores > div > svg {
  width: 60px;
}

.scores > div > h3 {
  margin: 0;
}