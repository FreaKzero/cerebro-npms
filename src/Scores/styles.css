.scores {
  display: flex;
  width: 600px;
  max-width: 600px;
}

.scores > div {
  flex: 1;
  margin-right: 10px;
  padding: 5px;
  position: relative;
  text-align: center;
}

.scores > div > h2 {
  position: absolute;
  left: 80px;
  top: 73px;
}

.scores > div > svg {
  width: 100px;
}

.scores > div > h3 {
  margin: 0;
}