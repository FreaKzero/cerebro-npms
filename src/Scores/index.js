import React, { Component, PropTypes } from 'react';
import {Circle} from 'rc-progress';
import {getColor} from '../lib';
import styles from './styles.css'

export default ({ popularity, maintenance, quality }) => (
    <div className={styles.scores}>
      <div>
        <p>Quality</p>
        <h2>{quality}</h2>
        <Circle
          percent={quality}
          strokeWidth="10"
          strokeColor={getColor(quality)}
        />
      </div>
      <div>
        <p>Popularity</p>
        <h2>{popularity}</h2>
        <Circle
          percent={popularity}
          strokeWidth="10"
          strokeColor={getColor(popularity)}
        />
      </div>
      <div>
        <p>Maintenance</p>
        <h2>{maintenance}</h2>
        <Circle
          percent={maintenance}
          strokeWidth="10"
          strokeColor={getColor(maintenance)}
        />
      </div>
    </div>
)