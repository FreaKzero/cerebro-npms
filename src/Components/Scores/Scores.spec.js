import React from 'react';
import Scores from './index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {   

  const scores ={
    quality: 0.9392171854054061,
    popularity: 0.06060838884010745,
    maintenance: 0.9999999595107407
  };
  
    const actions = {
      open: () => { return; },
      hideWindow: () => { return; }
    }

  const tree = renderer
    .create(<Scores {...scores}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});