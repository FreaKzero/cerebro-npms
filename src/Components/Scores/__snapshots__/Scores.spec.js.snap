// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={undefined}
>
  <div>
    <p>
      Quality
    </p>
    <h4
      style={
        Object {
          "color": "#d41243",
        }
      }
    >
      0.9392171854054061
      %
    </h4>
    <svg
      className="rc-progress-circle "
      style={Object {}}
      viewBox="0 0 100 100"
    >
      <path
        className="rc-progress-circle-trail"
        d="M 50,50 m 0,-45
           a 45,45 0 1 1 0,90
           a 45,45 0 1 1 0,-90"
        fillOpacity="0"
        stroke="#D9D9D9"
        strokeWidth={1}
        style={
          Object {
            "strokeDasharray": "282.7433388230814px 282.7433388230814px",
            "strokeDashoffset": "-0px",
            "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
          }
        }
      />
      <path
        className="rc-progress-circle-path"
        d="M 50,50 m 0,-45
           a 45,45 0 1 1 0,90
           a 45,45 0 1 1 0,-90"
        fillOpacity="0"
        stroke="#d41243"
        strokeLinecap="butt"
        strokeWidth="10"
        style={
          Object {
            "strokeDasharray": "2.655574028815416px 282.7433388230814px",
            "strokeDashoffset": "-0px",
            "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
          }
        }
      />
    </svg>
  </div>
  <div>
    <p>
      Maintenance
    </p>
    <h4
      style={
        Object {
          "color": "#d41243",
        }
      }
    >
      0.9999999595107407
      %
    </h4>
    <svg
      className="rc-progress-circle "
      style={Object {}}
      viewBox="0 0 100 100"
    >
      <path
        className="rc-progress-circle-trail"
        d="M 50,50 m 0,-45
           a 45,45 0 1 1 0,90
           a 45,45 0 1 1 0,-90"
        fillOpacity="0"
        stroke="#D9D9D9"
        strokeWidth={1}
        style={
          Object {
            "strokeDasharray": "282.7433388230814px 282.7433388230814px",
            "strokeDashoffset": "-0px",
            "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
          }
        }
      />
      <path
        className="rc-progress-circle-path"
        d="M 50,50 m 0,-45
           a 45,45 0 1 1 0,90
           a 45,45 0 1 1 0,-90"
        fillOpacity="0"
        stroke="#d41243"
        strokeLinecap="butt"
        strokeWidth="10"
        style={
          Object {
            "strokeDasharray": "2.82743327375013px 282.7433388230814px",
            "strokeDashoffset": "-0px",
            "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
          }
        }
      />
    </svg>
  </div>
  <div>
    <p>
      Popularity
    </p>
    <h4
      style={
        Object {
          "color": "#d41243",
        }
      }
    >
      0.06060838884010745
      %
    </h4>
    <svg
      className="rc-progress-circle "
      style={Object {}}
      viewBox="0 0 100 100"
    >
      <path
        className="rc-progress-circle-trail"
        d="M 50,50 m 0,-45
           a 45,45 0 1 1 0,90
           a 45,45 0 1 1 0,-90"
        fillOpacity="0"
        stroke="#D9D9D9"
        strokeWidth={1}
        style={
          Object {
            "strokeDasharray": "282.7433388230814px 282.7433388230814px",
            "strokeDashoffset": "-0px",
            "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
          }
        }
      />
      <path
        className="rc-progress-circle-path"
        d="M 50,50 m 0,-45
           a 45,45 0 1 1 0,90
           a 45,45 0 1 1 0,-90"
        fillOpacity="0"
        stroke="#d41243"
        strokeLinecap="butt"
        strokeWidth="10"
        style={
          Object {
            "strokeDasharray": "0.17136618221339564px 282.7433388230814px",
            "strokeDashoffset": "-0px",
            "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
          }
        }
      />
    </svg>
  </div>
</div>
`;
